content/blog/tidymodels-2025-q1/index.Rmd

```diff
@@ -0,0 +1,235 @@
+
+```
+Error in `fit()`:
+! The `formula` argument must be a formula, but it is a <character>.
+```
+
+and now you see
+
+```
+Error in `fit()`:
+! `formula` must be a formula, not the string "boop".
+```
+
+This was _a lot_ of work and we're still isn’t finished. Two events helped us get as far as we did. 
+
+First, Simon Couch made the [chores](https://simonpcouch.github.io/chores/) package (its previous name was "pal"), which enabled us to use AI tools to solve small-scope problems, such as converting old rlang error code to use the new [cli syntax](https://rlang.r-lib.org/reference/topic-condition-formatting.html). I can’t overstate how much of a speed-up this was for us. 
+
+Second, at developer day, many external folks pitched in to make pull requests from a list of issues:
+
+```{r}
+#| label: post-its
+#| fig.cap: Organizing Tidy Dev Day issues.
+#| echo: false
+#| fig.align: center
+
+knitr::include_graphics("IMG_4743.jpeg")
+```
+
+I love these sessions for many reasons, but mostly because we meet users and contributors to our packages in person and work with them on specific tasks. 
+
+There is a lot more to do here; we have a lot of secondary packages that would benefit from these improvements too. 
+
+## The parsnip Package
+
+One big update in parsnip was a new modeling mode of `"quantile regression"`. Daniel McDonald and Ryan Tibshirani largely provided some inertia for this work based on their disease modeling framework. 
+
+You can generate quantile predictions by first creating a model specification, which includes the quantiles that you want to predict:
+
+```{r}
+#| label: sshhh
+#| include: false
+library(tidymodels)
+```
+
+```{r}
+#| label: quant-spec
+library(tidymodels)
+tidymodels_prefer()
+
+ames <- 
+  modeldata::ames |> 
+  mutate(Sale_Price = log10(Sale_Price)) |> 
+  select(Sale_Price, Latitude)
+
+quant_spec <- 
+  linear_reg() |> 
+  set_engine("quantreg") |> 
+  set_mode("quantile regression", quantile_levels = c(0.1, 0.5, 0.9))
+quant_spec
+```
+
+We'll add some spline terms via a recipe and fit the model:
+
+```{r}
+#| label: quant-fit
+spline_rec <- 
+  recipe(Sale_Price ~ ., data = ames) |> 
+  step_spline_natural(Latitude, deg_free = 10)
+
+quant_fit <- 
+  workflow(spline_rec, quant_spec) |> 
+  fit(data = ames)
+
+quant_fit
+```
+
+For prediction, tidymodels always returns a data frame with as many rows as the input data set (i.e., `ames`). The result is a special vctrs class: 
+
+```{r}
+#| label: quant-pred
+quant_pred <- predict(quant_fit, ames) 
+quant_pred |> slice(1:4)
+
+class(quant_pred$.pred_quantile)
+```
+
+where the output `[5.31]` shows the middle quantile. 
+
+We can expand the set of quantile predictions so that there are three rows for each source row in `ames` . There’s also an integer column called `.row` so that we can merge the data with the source data:  
```

Comment:
```suggestion
We can expand the set of quantile predictions so that there are three rows for each source row in `ames`. There’s also an integer column called `.row` so that we can merge the data with the source data:  
```